### PR TITLE
Add Twitter Media api support

### PR DESCRIPTION
**Pull request by ryankdwyer**

### CODE REVIEW

File: lib/Net/Twitter/Role/API/Media.pm [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+use Moose::Role;
+use Net::Twitter::API;
+use DateTime::Format::Strptime;
+use namespace::autoclean;
+
+
+has media_api_url => (
+    isa => 'Str',
+    is => 'rw',
```
Comment (skaufman-socialflow) [inline lib/Net/Twitter/Role/API/Media.pm]: why rw vs ro?

File: lib/Net/Twitter/Role/API/Media.pm [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    default => 'http://api.twitter.com/1.1',
+);
```
Comment (skaufman-socialflow) [inline lib/Net/Twitter/Role/API/Media.pm]: Why specify this vs just using the one specified in `Net::Twitter::Role::API::RESTv1_1`?
Could you just call
`with "Net::Twitter::Role::API::RESTv1_1";` instead?

**Review by skaufman-socialflow — COMMENTED** [review]
had some minor feedback

**Comment by skaufman-socialflow** [comment]
@ryankdwyer I think this should be released as a separate dist, you can copy it to sf-labs/dist/ or something, but because it requires clearance from twitter and is an unadvertised API we should not release it as open source.

File: lib/Net/Twitter/Role/API/Media.pm (hunk 1 above)
Comment (ryankdwyer) [reply lib/Net/Twitter/Role/API/Media.pm]: media_api_url potentially gets altered in the BUILD subroutine

**Review by antipasta — APPROVED** [review]

**Pull request merged**